samples/tests/ComponentExamples/ComponentExamples.test.ts

```diff
@@ -0,0 +1,56 @@
+import { test, expect } from '@playwright/test';
+import { encodeQueryData, createChatThreadAndUsers } from '../common/utils';
+import * as path from 'path';
+import dotenv from 'dotenv';
+
+dotenv.config({ path: path.join(__dirname, '..', '.env') });
+
+const SERVER_URL = 'http://localhost:8080';
+const PARTICIPANTS = ['Jack'];
+
+const dataUiId = (id: string): string => `[data-ui-id="${id}"]`;
+
+test.describe('Component Examples Test', () => {
+  test.beforeEach(async ({ page }) => {
+    // Go to the starting url before each test.
+    const user = await createChatThreadAndUsers(PARTICIPANTS);
+
+    const qs = encodeQueryData(user[0]);
+    const url = `${SERVER_URL}?${qs}`;
+    await page.goto(url);
+  });
+
+  // only need to make sure it is loading correctly, the rest will be the responsibility of composite tests
+  test('Whether html page is loaded right with 2 composites', async ({ page }) => {
+    expect(await page.waitForSelector('text=Connect components to StatefulClient')).toBeTruthy();
```

Comment:
Can you replace these selectors with injected `data-ui-id` tags. The uniform way to access elements helps readability.